assets/js/plugin-check-admin.js

```diff
@@ -25,17 +27,96 @@
 		);
 	} );
 
+	/**
+	 * Setup the runtime environment if needed.
+	 *
+	 * @since n.e.x.t
+	 */
+	function setupEnvironment( data ) {
+		const pluginCheckData = new FormData();
+		pluginCheckData.append( 'nonce', pluginCheck.nonce );
+		pluginCheckData.append( 'plugin', data.plugin );
+		pluginCheckData.append( 'action', 'plugin_check_set_up_environment' );
+
+		for (var i = 0; i < data.checks.length; i++) {
+			pluginCheckData.append('checks[]', data.checks[i]);
+		}
+
+		return fetch(
+			ajaxurl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				return response.json();
+			}
+		)
+		.then( handleDataErrors )
+		.then(
+			( data ) => {
+				if ( ! data.data || ! data.data.message ) {
+					throw new Error( 'Response contains no data.' );
+				}
+
+				console.log( data.data.message );
+
+				return data.data;
+			}
+		);
+	}
+
+	/**
+	 * Cleanup the runtime environment.
+	 *
+	 * @since n.e.x.t
+	 */
+	function cleanupEnvironment( data ) {
+		const pluginCheckData = new FormData();
+		pluginCheckData.append( 'nonce', pluginCheck.nonce );
+		pluginCheckData.append( 'action', 'plugin_check_clean_up_environment' );
+
+		return fetch(
+			ajaxurl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				return response.json();
+			}
+		)
+		.then( handleDataErrors )
+		.then(
+			( data ) => {
+				if ( ! data.data || ! data.data.message ) {
+					throw new Error( 'Response contains no data.' );
+				}
+
+				console.log( data.data.message );
+
+				return data.data;
+			}
+		);
+	}
+
+
 	/**
 	 * Get the Checks to run.
 	 *
 	 * @since n.e.x.t
 	 */
 	function getChecksToRun() {
-		// Collect the data to pass along for generating a check results.
 		const pluginCheckData = new FormData();
 		pluginCheckData.append( 'nonce', pluginCheck.nonce );
 		pluginCheckData.append( 'plugin', pluginsList.value );
-		pluginCheckData.append( 'checks', [] );
+		pluginCheckData.append( 'checks[]', [] );
```

Comment:
Doesn't this parameter need to not be set at all for it to be interpreted as an empty array? If that causes an error, we may need to adjust the server-side logic to ensure that the `checks` parameter is not _required_ (i.e. can be an empty array).

But setting this here to an empty array technically means this is an array with one element that is itself an array. So that wouldn't be right.